### PR TITLE
MSEARCH-325 Item search/facet/sort options: Support 'item' references (backward compatibility)

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
### Purpose
Add SearchQueryModifiers, that modify query before search

#### TODOS and Open Questions
- Write Karate tests 
- Remove backward compatibility (item query modifier)

### Learning
https://issues.folio.org/browse/MSEARCH-325

### CODE REVIEW

File: src/main/resources/model/instance.json [hunk 1]
```diff
@@ -220,36 +220,46 @@
...
         "status": {
+          "searchAliases": "item.status",
           "type": "object",
           "properties": {
             "name": {
               "searchTypes": [ "facet", "filter", "sort" ],
+              "searchAliases": "item.status.name",
               "index": "keyword",
               "sort": {
                 "fieldName": "items.status.name",
```
Comment (Shans-Kaluhin) [inline src/main/resources/model/instance.json:265]: It's no way to add searchAliases or second fieldName for sort parameter.
So for now I added `sort_item.status.name` field, and Processor for this filed that should be removed with renaming items
```
 "sort_item.status.name": {
      "searchTypes": "sort",
      "type": "search",
      "index": "lowercase_sort",
      "processor": "sortItemStatusProcessor"
    },
```

File: src/main/java/org/folio/search/service/setter/item/SortItemStatusProcessor.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.search.domain.dto.Instance;
+import org.folio.search.service.setter.FieldProcessor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/setter/item/SortItemStatusProcessor.java:12]: It's not used there

**Review by pfilippov-epam — APPROVED** [review]
Looks good

File: src/main/java/org/folio/search/cql/ItemSearchFieldModifier.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.folio.search.cql;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ItemSearchFieldModifier implements SearchFieldModifier {
+
+  @Override
+  public String modify(String inputField) {
+    return inputField.replace("item.", "items.");
```
Comment (psmagin) [inline src/main/java/org/folio/search/cql/ItemSearchFieldModifier.java:10]: I think that safer to add a check that `inputField` starts with "item." and only then replace it with "items."

File: README.md [hunk 4]
```diff
@@ -402,7 +402,34 @@ if it is defined but doesn't match.
...
+| `itemPublicNotes`                   | full-text | `itemPublicNotes all "public note"`                          | Search by item public notes and circulation notes                                                                             |
+| `itemIdentifiers`                   |   term    | `itemIdentifiers all "81ae0f60-f2bc-450c-84c8-5a21096daed9"` | Search by item Identifiers: `item.id`, `item.hrid`, `item.formerIds`, `item.accessionNumber`, `item.itemIdentifier`      |
+| `item.metadata.createdDate`         |   term    | `metadata.createdDate > "2020-12-12"`                        | Matches instances with item that were created after  `2020-12-12`                                                            |
+| `item.metadata.updatedDate`         |   term    | `metadata.updatedDate > "2020-12-12"`                        | Matches instances with item that were updated after  `2020-12-12`                                                            |
+
+### Items search options (Backward compatibility - should be removed in next release)
```
Comment (psmagin) [inline README.md:432]: I don't think we will remove it.

File: doc/development.md [hunk 5]
```diff
@@ -16,17 +16,18 @@ the [full-text queries](https://www.elastic.co/guide/en/elasticsearch/reference/
...
+| eventBodyJavaClass   | The Java class that incoming JSON can be mapped to. Currently, it's used to make processing of search field more convenient                                                                                                                                                                     |
+| languageSourcePaths  | Contains a list of json path expressions to extract languages values in ISO-639 format. If the multi-language is supported for the resource, this path must be specified.                                                                                                                       |
+| searchFieldModifiers | Contains list of field modifiers, which pre-processes incoming fields for elasticsearch request.                                                                                                                                         |
```
Comment (psmagin) [inline doc/development.md:25]:
```suggestion
| searchFieldModifiers | Contains a list of field modifiers, which pre-processes incoming fields for elasticsearch request.                                                                                                                                         |
```

File: src/test/java/org/folio/search/cql/ItemSearchFieldModifierTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  private ItemSearchFieldModifier itemSearchFieldModifier;
+
+  @BeforeEach
+  void setUp() {
+    itemSearchFieldModifier = new ItemSearchFieldModifier();
+  }
+
+  @Test
+  void getSearchTerm_positive() {
+    var field = "items.field";
```
Comment (psmagin) [inline src/test/java/org/folio/search/cql/ItemSearchFieldModifierTest.java:21]:
```suggestion
    var field = "item.field";
```
Maybe this is a case?

**Review by psmagin — APPROVED** [review]

**Pull request merged**